dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/JsonUpdateTest.java

```diff
@@ -0,0 +1,190 @@
+        String updateValue = "value3";
+        String actual = jsonUpdate.eval(jsonEvent, jPath, updateValue);
+        Assert.assertEquals(expectedJsonEvent, actual);
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNestedJson() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":{\"key1\":\"value1\",\"key2\":\"value2\"}}";
+        String jPath = "$.k2";
+        Map<String, String> updateValue = new HashMap<String, String>() {{
+            put("key1", "value1");
+            put("key2", "value2");
+        }};
+        Assert.assertEquals(expectedJsonEvent, jsonUpdate.eval(jsonEvent, jPath, updateValue));
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNullValue() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":null}";
+        String jPath = "$.k2";
+        String updateValue = null;
+        Assert.assertEquals(expectedJsonEvent, jsonUpdate.eval(jsonEvent, jPath, updateValue));
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNewNestedKeyValue() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"key1\":\"value1\",\"key2\":\"value2\"}}";
+        String jPath = "$.k3";
+        Map<String, String> updateValue = new HashMap<String, String>() {{
+            put("key1", "value1");
+            put("key2", "value2");
+        }};
+        String actual = jsonUpdate.eval(jsonEvent, jPath, updateValue);
+        Assert.assertEquals(expectedJsonEvent, actual);
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNewNestedKeyValueInNestedPath() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"nk1\":\"nv1\",\"nk2\":\"nv2\"}}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"nk1\":{\"key1\":\"value1\",\"key2\":\"value2\"},\"nk2\":\"nv2\"}}";
+        String jPath = "$.k3.nk1";
+        Map<String, String> updateValue = new HashMap<String, String>() {{
+            put("key1", "value1");
+            put("key2", "value2");
+        }};
+        String actual = jsonUpdate.eval(jsonEvent, jPath, updateValue);
+        Assert.assertEquals(expectedJsonEvent, actual);
+    }
+
+    @Test
+    public void shouldThrowErrorWhenNullJsonEvent() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("json string can not be null or empty");
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = null;
+        String jPath = "$.k2";
+        String updateValue = "updatedValue";
+        jsonUpdate.eval(jsonEvent, jPath, updateValue);
+    }
+
```

Comment:
What error is thrown when jsonEvent is not a proper JSON string ?

Reply:
Can we have a test for it as well if possible ?

Reply:
It will through `com.jayway.jsonpath.InvalidJsonException` and the test case for the same is added.

Reply:
ok